Assets/PlayerStates.cs

```diff
@@ -0,0 +1,24 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class PlayerStates : MonoBehaviour {
+
+    Health playerHealth;
+    Stamina playerStamina;
+	// Use this for initialization
+	void Start ()
+    {
+        playerHealth = GetComponent<Health>();
+        playerStamina = GetComponent<Stamina>();
+
+	}
+
+    public void death ()
+    {
+        //respawn, reset hp/stamina, etc.
+        playerHealth.add(playerHealth.getMaxHealth());
```

Comment:
have playerHealth and playerStamina have a .Reset() method that does this interally, instead of adding the maximum health back. then call playerHealth.Rest(). playerStamina.Reset(). See previous comment about a base class for these two that might want to implement Reset().